operator/pkg/reconciliation/constructor.go

```diff
@@ -104,18 +114,52 @@ func newNamespacedNameForStatefulSet(
 	}
 }
 
-// Create a statefulset object for the Datacenter.
+func newStatefulSetForCassandraDatacenterWithDefunctPvcManagedBy(
```

Comment:
maybe add a comment for why this is here - your other comment is really great - just edited the intro...
```
// We have to account for the fact that they might use the old managed-by label value
// (oplabels.ManagedByLabelDefunctValue) for CassandraDatacenters originally
// created in version 1.1.0 or earlier.
```